security/pkg/nodeagent/sds/sdsservice.go

```diff
@@ -437,6 +441,41 @@ func (s *sdsservice) FetchSecrets(ctx context.Context, discReq *discovery.Discov
 	return sdsDiscoveryResponse(secret, resourceName, discReq.TypeUrl)
 }
 
+func (s *sdsservice) getToken() (string, error) {
+	token := ""
+	needRenew := false
+	tok, err := ioutil.ReadFile(s.jwtPath)
+	if err != nil {
+		sdsServiceLog.Errorf("failed to get credential token: %v from path %s", err, s.jwtPath)
+		needRenew = true
+	} else {
+		tokenExpired, err := util.IsJwtExpired(string(tok), time.Now())
+		if err != nil || tokenExpired {
+			sdsServiceLog.Errorf("JWT expiration checking error: %v or token is expired %v", err, tokenExpired)
+			needRenew = true
+		} else {
+			// We have a valid token.
+			token = string(tok)
+			return token, nil
+		}
+	}
+	if needRenew {
+		// TODO (liminw): CredFetcher is a general interface. In 1.7, we limit the use on GCE only because
+		// GCE is the only supported plugin at the moment.
+		if s.credFetcher != nil && s.credFetcher.GetType() == security.GCE {
```

Comment:
The logic should be: 
- if s.credFetchoer != nil  - use it ( no need to read the jwtPath ).
- else - use the jwtPath.

Reply:
If there is existing valid token, we do not need to call metadata server to get a new one. This can save unnecessary calls to GCE metadata server.